java-frontend/src/main/java/org/sonar/java/se/checks/ConditionAlwaysTrueOrFalseCheck.java

```diff
@@ -48,14 +51,21 @@ public void checkEndOfExecution(CheckerContext context) {
     EvaluatedConditions ec = evaluatedConditions.pop();
     for (Tree condition : Sets.difference(ec.evaluatedToFalse.keySet(), ec.evaluatedToTrue.keySet())) {
       context.reportIssue(condition, this, "Change this condition so that it does not always evaluate to \"false\"",
-        ec.evaluatedToFalse.get(condition).stream().map(node -> FlowComputation.flow(node.parent(), node.programState.peekValue())).collect(Collectors.toSet()));
+        collectFlow(condition, ec.evaluatedToFalse, false));
     }
     for (Tree condition : Sets.difference(ec.evaluatedToTrue.keySet(), ec.evaluatedToFalse.keySet())) {
       context.reportIssue(condition, this, "Change this condition so that it does not always evaluate to \"true\"",
-        ec.evaluatedToTrue.get(condition).stream().map(node -> FlowComputation.flow(node.parent(), node.programState.peekValue())).collect(Collectors.toSet()));
+        collectFlow(condition, ec.evaluatedToTrue, true));
     }
   }
 
+  private static Set<List<JavaFileScannerContext.Location>> collectFlow(Tree condition, Multimap<Tree, ExplodedGraph.Node> nodes, boolean conditionIsAlwaysTrue) {
+    Set<List<JavaFileScannerContext.Location>> flows = nodes.get(condition).stream().map(node -> FlowComputation.flow(node.parent(), node.programState.peekValue()))
+      .collect(Collectors.toSet());
+    flows.forEach(f -> f.add(new JavaFileScannerContext.Location("Condition is always " + conditionIsAlwaysTrue, condition)));
```

Comment:
You can probably define a dedicated function to append instead of collecting and then doing a for each.